src/realm/sync/subscriptions.hpp

```diff
@@ -157,6 +158,8 @@ class SubscriptionSet {
     // subscription set will be unchanged.
     SubscriptionSet make_mutable_copy() const;
 
+    util::Future<void> get_state_change_notification(State notify_when) const;
```

Comment:
Should this accept a `State` parameter? Right now we have 3 states and the subscription will always move from `Pending` to either `Complete` or `Error` - I would expect that SDKs will want to be notified of the change either way so that they can either resolve or reject the SDK promise.

Reply:
Sure, we can do that.